mypy_django_plugin/transformers/managers.py

```diff
@@ -183,112 +247,198 @@ def create_new_manager_class_from_from_queryset_method(ctx: DynamicClassDefConte
+        return
+
+    assert isinstance(manager_base, TypeInfo)
+
+    callee = ctx.call.callee
+    assert isinstance(callee, MemberExpr)
+    assert isinstance(callee.expr, RefExpr)
+
+    queryset_info = callee.expr.node
+    if queryset_info is None:
+        if not semanal_api.final_iteration:
+            semanal_api.defer()
+        return
+
+    assert isinstance(queryset_info, TypeInfo)
+
+    manager_class_name = manager_base.name + "From" + queryset_info.name
+    current_module = semanal_api.modules[semanal_api.cur_mod_id]
+    existing_sym = current_module.names.get(manager_class_name)
+    if (
+        existing_sym is not None
+        and isinstance(existing_sym.node, TypeInfo)
+        and existing_sym.node.has_base(fullnames.MANAGER_CLASS_FULLNAME)
+        and existing_sym.node.metadata.get("django", {}).get("from_queryset_manager") == queryset_info.fullname
+    ):
+        # Reuse an identical, already generated, manager
+        new_manager_info = existing_sym.node
+    else:
+        manager_base_instance = fill_typevars(manager_base)
+        assert isinstance(manager_base_instance, Instance)
+        new_manager_info = helpers.add_new_class_for_module(
+            module=current_module,
+            name=manager_class_name,
+            bases=[manager_base_instance],
+        )
+        new_manager_info.type_vars = manager_base.type_vars
+        new_manager_info.line = ctx.call.line
+        new_manager_info.defn.type_vars = manager_base.defn.type_vars
+        new_manager_info.defn.line = ctx.call.line
+
+        merge_queryset_into_manager(
+            api=semanal_api,
+            manager_info=new_manager_info,
+            queryset_info=queryset_info,
+            manager_base=manager_base,
+            class_name=manager_class_name,
         )
 
-    # Insert the new manager (dynamic) class
-    assert semanal_api.add_symbol_table_node(ctx.name, SymbolTableNode(GDEF, new_manager_info, plugin_generated=True))
+    # Whenever `<QuerySet>.as_manager()` isn't called at class level, we want to ensure
+    # that the variable is an instance of our generated manager. Instead of the return
+    # value of `.as_manager()`. Though model argument is populated as `Any`.
+    # `transformers.models.AddManagers` will populate a model's manager(s), when it
+    # finds it on class level.
+    var = Var(name=ctx.name, type=Instance(new_manager_info, [AnyType(TypeOfAny.from_omitted_generics)]))
+    var.info = new_manager_info
+    var._fullname = f"{current_module.fullname}.{ctx.name}"
+    var.is_inferred = True
+    # Note: Order of `add_symbol_table_node` calls matters. Depending on what level
+    # we've found the `.as_manager()` call. Point here being that we want to replace the
+    # `.as_manager` return value with our newly created manager.
+    assert semanal_api.add_symbol_table_node(
+        ctx.name, SymbolTableNode(semanal_api.current_symbol_kind(), var, plugin_generated=True)
+    )
```

Comment:
```suggestion
    symbol_kind = semanal_api.current_symbol_kind()
     if ctx.name != new_manager_info.name:
         # Unless names are equal, there's 2 symbol names that needs the manager info
         assert semanal_api.add_symbol_table_node(
             ctx.name, SymbolTableNode(symbol_kind, new_manager_info, plugin_generated=True)
         )
```

Should we do the same here for good measure? I assume if it's a problem with `from_queryset()` it likely is with `as_manager()` as well?

Reply:
Yeah, lets not gamble

Reply:
Hm, just realised that this will be incorrect. Since `.as_manager` returns an _instance_ and not a type (which `.from_queryset` does). So in one way or another, we'll need a _node_ with a set type.